src/main/java/com/microsoft/sqlserver/jdbc/SQLCollation.java

```diff
@@ -587,9 +590,9 @@ final Encoding checkSupported() throws UnsupportedEncodingException
             // This works for all of the code pages above in SE 5 and later.
             try
             {
-                " ".getBytes(charsetName);
+                charset = Charset.forName(charsetName);
```

Comment:
instead of assigning the charset object to charset variable, is it possible to do the assignment outside the checkSupported() method? e.g. we could do the assignment after checkSupported() is called in charset()